src/server/webhook-events.ts

```diff
@@ -1,29 +1,77 @@
+import gql from 'graphql-tag';
+
+import {StatusToPRQuery} from '../types/gql-types';
+import {GitHub} from '../utils/github';
+
 import {sendNotification} from './controllers/notifications';
 import {DashSecrets} from './dash-server';
+import {userModel} from './models/userModel';
 
-type UserHook = {
+type User = {
   login: string;
 };
 
 type ReviewHook = {
-  state: 'approved'|'changes_requested'|'commented', user: UserHook;
+  state: 'approved'|'changes_requested'|'commented', user: User;
 };
 
 type PullRequestHook = {
-  title: string; user: UserHook;
+  title: string; user: User;
 };
 
 type RepositoryHook = {
-  name: string;
+  name: string; full_name: string;
 };
 
 type PullRequestReviewHook = {
   action: string; review: ReviewHook; pull_request: PullRequestHook;
   repository: RepositoryHook;
 };
 
+type StatusHook = {
+  sha: string; state: 'error' | 'failure' | 'pending' | 'success';
+  description: string;
+  repository: RepositoryHook;
+  commit: {author: User;}
+};
+
 // Triggered when the status of a Git commit changes.
-export async function handleStatus(_hookBody: string) {
+export async function handleStatus(
+    secrets: DashSecrets, github: GitHub, hookData: StatusHook) {
+  if (hookData.state !== 'error' && hookData.state !== 'failure') {
+    return;
+  }
+
+  // There was an error that should be reported to the PR Owner
+  const repo = hookData.repository;
+  const author = hookData.commit.author;
+
+  const loginDetails = userModel.getUserDetails(author.login);
+  if (!loginDetails) {
+    // We have no details so can't get the PR for the commit
+    return;
+  }
+
+  const prData = await github.query<StatusToPRQuery>({
+    query: statusToPRQuery,
+    variables: {
+      queryString: `type:pr repo:${repo.full_name} ${hookData.sha}`,
+    },
+    context: {token: loginDetails.token}
+  });
+  const nodes = prData.data.search.nodes;
+  if (nodes && nodes.length > 0) {
```

Comment:
Is it possible this SHA belongs to multiple PRs? eg. when I chain changes, and they all contain the same commit, but are in separate PRs to merge to master.

Reply:
It is possible - not sure of the scenario of this and what the expected behavior should be.

We could alter the message from `[<repo name>] <pull request name>` to `[<repo name>] <X PR Count>'s are affected by this status.`

Reply:
FTR, we discussed offline. This should probably fetch the last commits of the matching pull requests, and send a notification for each pull request that has this commit as the latest commit. Otherwise, we would be spamming on old commits for the PR.